src/modules/table/services/table.service.ts

```diff
@@ -58,24 +59,44 @@ export class TableService {
     const newTable = new TableEntity();
     Object.assign(newTable, createTableDto);
     newTable.restaurant = restaurant;
+    newTable.floor = floor;
     return await this.tableRepository.save(newTable);
   }
 
   async getByUser(query: TableQueryParams) {
     return this.tableRepository
       .createQueryBuilder('table')
-      .innerJoinAndSelect('table.user', 'user')
-      .innerJoinAndSelect('table.floor', 'floor')
+      .innerJoin('table.floor', 'floor')
+      .innerJoin('table.restaurant', 'restaurant')
+      .innerJoin('restaurant.user', 'user')
       .where('user.id = :userId', { userId: query.userId })
       .getMany();
   }
 
-  async deleteByUser(query: TableQueryParams) {
+  async getById(tableId: number) {
     return this.tableRepository
-      .createQueryBuilder() //Створення нового об'єкта QueryBuilder для виконання SQL-запитів.
-      .delete() //Вказуємо, що це буде операція DELETE.
-      .from(TableEntity) //Визначення таблиці, з якої видаляємо дані. TableEntity - це клас сутності, який представляє таблицю "table" в базі даних.
-      .where('user.id = :userId', { userId: query.userId }) //Умова видалення → для рядків, де значення стовпця "user.id" дорівнює query.userId.
-      .execute(); //Запуск SQL-запиту на виконання. В даному випадку, це виконає видалення рядків у вказаних умовах.
+      .createQueryBuilder('table')
+      .innerJoin('table.floor', 'floor')
+      .innerJoin('table.restaurant', 'restaurant')
+      .innerJoin('restaurant.user', 'user')
+      .where('table.id = :tableId', { tableId })
+      .getOne();
+  }
+
+  async delete(deleteTableDto: DeleteTableRequestDto, currentUserId: number) {
+    const errorHelper = new ErrorHelper();
+    const table = await this.getById(deleteTableDto.id);
+
+    console.log('table >>>>>>', table);
+
+    if (!table) {
+      errorHelper.addNewError(
+        `Table with given id:${deleteTableDto.id} does not exist`,
+        'restaurant',
```

Comment:
replace 'restaurant' to 'table'. If table won't find at database frontend take error object like this
```
{
  'table': 'Table with given id:${deleteTableDto.id} does not exist'
}
```
It need because some time there more then one argument to check.

Reply:
Fixed